libsentrykube/quickpatch.py

```diff
@@ -1,15 +1,142 @@
-from typing import Sequence, Mapping
+from pathlib import Path
+import re
+from typing import MutableMapping, Sequence
+import click
+import yaml
+import jsonpatch
+
+from libsentrykube.service import (
+    get_tools_managed_service_value_overrides,
+    get_service_path,
+    write_managed_values_overrides,
+)
+from jsonschema import validate, ValidationError
+
+
+def find_patch_file(service: str, patch: str) -> Path:
+    """
+    Finds the patch file for the given service and patch name
+    """
+    try:
+        base_path = get_service_path(service)
+    except click.Abort:
+        raise FileNotFoundError(f"Service {service} not found")
+    expected_path = Path(base_path) / "quickpatches"
+
+    if expected_path.exists() and expected_path.is_dir():
+        patch_file = expected_path / f"{patch}.yaml"
+        if patch_file.exists():
+            return patch_file
+    raise FileNotFoundError(f"Patch file {patch}.yaml not found")
+
+
+def load_and_validate_yaml(file_path: Path, patch: str) -> dict:
+    """
+    Load the patch file and validate for required fields
+    """
+    try:
+        with open(file_path, "r") as file:
+            patch_data = yaml.safe_load(file)
+    except yaml.YAMLError as e:
+        raise ValueError(f"Invalid yaml in patch file {patch}.yaml: {e}") from e
+    if "mappings" not in patch_data:
+        raise ValueError(f"Resource mappings not found in patch file {patch}.yaml")
+    if "schema" not in patch_data:
+        raise ValueError(f"Schema not found in patch file {patch}.yaml")
+    if "patches" not in patch_data:
+        raise ValueError(f"Patches not found in patch file {patch}.yaml")
+
+    # Validate the schema
+    schema = patch_data.get("schema")
+    if (
+        "additionalProperties" not in schema
+        or schema["additionalProperties"] is not False
+    ):
+        raise ValueError(
+            f"Schema additionalProperties must be False in patch file {patch}.yaml"
+        )
+
+    # Find all variables enclosed in <> in the file content
+    file_content = file_path.read_text()
+    variables = set(re.findall(r"<\s*(\w+)\s*>", file_content))
```

Comment:
Please add this `(?<!/)` to ignore the `<` if it is escaped `/<` 
`(?<!/)<\s*(\w+)\s*>`

Also the regex is missing the `-` as `\w` does not include it.